src/lib/gundb/UserStorage.js

```diff
@@ -700,23 +715,25 @@ export class UserStorage {
       if (insertPos >= 0) dayEventsArr.splice(insertPos, 0, eventIndexItem)
       else dayEventsArr.unshift(eventIndexItem)
     }
-
     let saveAck = this.feed
       .get(day)
       .putAck(JSON.stringify(dayEventsArr))
-      .catch(err => logger.error(err))
+      .catch(err => logger.error('updateFeedEvent dayIndex', err))
     const ack = this.feed
       .get('index')
       .get(day)
       .putAck(dayEventsArr.length)
+      .catch(err => logger.error('updateFeedEvent daySize', err))
 
     if (event.data && event.data.receipt) {
       await this.saveLastBlockNumber(event.data.receipt.blockNumber)
     }
 
     const result = await Promise.all([saveAck, ack])
-      .then(arr => arr[0])
-      .catch(err => logger.info('savingIndex', err))
+      .then(arr => {
+        return event
```

Comment:
so `result` is `event` or `undefined`?

you can change this to:
```javascript
await Pormise.all([saveAck, ack]).catch(err => logger.error('savingIndex', err))
```
Neither the `result` `updateFeedEvent` call is expected. It's jus waiting for the promises to be resolved.

Function signature changes as well, something in the order of `Promise<>`

Reply:
changed signature to Promise<FeedEvent>

Reply:
you always want to return the `event`?

then change the block:
```javascript
    const result = await Promise.all([saveAck, ack])
      .then(arr => {
        return event
      })
      .catch(err => logger.error('savingIndex', err))
    return result
```

to:
```javascript
return Promise.all([saveAck, ack])
  .catch(err => logger.error('savingIndex', err))
  .then(() => event)
```

if you don't care about the returned value. Then go with the first approach I metioned.